modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php

```diff
@@ -0,0 +1,322 @@
+					'_en': 'woocommerceanalytics_add_to_cart',
+					'blog_id': " . esc_js( $blogid ) . ",
+					'pi': '" . esc_js( $product->get_id() ) . "',
+					'pn' : '" . esc_js( $product->get_title() ) . "',
+					'pq': jQuery( 'input.qty' ).val() ? jQuery( 'input.qty' ).val() : '1',
+					'ui': '" . esc_js( $this->get_user_id() ) . "',
+				} );
+			} );"
+		);
+	}
+
+	/**
+	 * On product lists or other non-product pages, add an event listener to "Add to Cart" button click
+	 */
+	public function loop_add_to_cart() {
+		$blogid   = Jetpack::get_option( 'id' );
+		$selector = '.add_to_cart_button:not(.product_type_variable, .product_type_grouped)';
+
+		wc_enqueue_js(
+			"jQuery( '" . esc_js( $selector ) . "' ).click( function() {
+				var productID = jQuery( this ).data( 'product_id' );
+				var productDetails = {
+					'id': productID,
+					'quantity': jQuery( this ).data( 'quantity' ),
+				};
+				_wca.push( {
+					'_en': 'woocommerceanalytics_product_view',
+					'blog_id': '" . esc_js( $blogid ) . "',
+					'pi': productDetails.id,
+					'ui': '" . esc_js( $this->get_user_id() ) . "',
+				} );
+				_wca.push( {
+					'_en': 'woocommerceanalytics_add_to_cart',
+					'blog_id': " . esc_js( $blogid ) . ",
+					'pi': productDetails.id,
+					'pq': productDetails.quantity,
+					'ui': '" . esc_js( $this->get_user_id() ) . "',
+				} );
+			} );"
+		);
+	}
+
+	/**
+	 * On the cart page, add an event listener for removal of product click
+	 */
+	public function remove_from_cart() {
+
+		// We listen at div.woocommerce because the cart 'form' contents get forcibly
+		// updated and subsequent removals from cart would then not have this click
+		// handler attached.
+		$blogid = Jetpack::get_option( 'id' );
+		wc_enqueue_js(
+			"jQuery( 'div.woocommerce' ).on( 'click', 'a.remove', function() {
+				var productID = jQuery( this ).data( 'product_id' );
+				var quantity = jQuery( this ).parent().parent().find( '.qty' ).val()
+				var productDetails = {
+					'id': productID,
+					'quantity': quantity ? quantity : '1',
+				};
+				_wca.push( {
+					'_en': 'woocommerceanalytics_remove_from_cart',
+					'blog_id': '" . esc_js( $blogid ) . "',
+					'pi': productDetails.id,
+					'pq': productDetails.quantity,
+					'ui': '" . esc_js( $this->get_user_id() ) . "',
+				} );
+			} );"
+		);
+	}
+
+	/**
+	 * Adds the product ID to the remove product link (for use by remove_from_cart above) if not present
+	 *
+	 * @param string $url url.
+	 * @param string $key key.
+	 * @return mixed.
+	 */
+	public function remove_from_cart_attributes( $url, $key ) {
+		if ( false !== strpos( $url, 'data-product_id' ) ) {
+			return $url;
+		}
+
+		$item    = WC()->cart->get_cart_item( $key );
+		$product = $item['data'];
+
+		$new_attributes = sprintf(
+			'href="%s" data-product_id="%s" data-product_sku="%s"',
+			esc_attr( $url ),
+			esc_attr( $product->get_id() ),
+			esc_attr( $product->get_sku() )
+		);
+		$url = str_replace( 'href=', $new_attributes );
```

Comment:
Isn't this missing a third parameter, the string where we want to find `'href='`?

Reply:
Yes! [It should](http://php.net/manual/en/function.str-replace.php). @allendav, FYI the same issue is here
https://github.com/Automattic/jetpack/blob/e1ebfd8d145e4759bee85c833c5d232fa3c893b4/modules/google-analytics/classes/wp-google-analytics-universal.php#L266

@greenafrican, is this code even needed?